mne_bids/tests/test_read.py

```diff
@@ -327,20 +330,69 @@ def test_handle_coords_reading():
     raw.set_montage(montage)
     write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
 
-    # read in the data and assert montage is the same
-    bids_fname = bids_basename + "_ieeg.edf"
+    # obtain the sensor positions and assert ch_coords are same
     raw_test = read_raw_bids(bids_fname, bids_root)
-
-    # obtain the sensor positions
     orig_locs = raw.info['dig'][1]
     test_locs = raw_test.info['dig'][1]
     assert orig_locs == test_locs
     assert not object_diff(raw.info['chs'], raw_test.info['chs'])
 
-    # test error message if electrodes don't match
+    # read in the data and assert montage is the same
+    # regardless of 'm', 'cm', 'mm', or 'pixel'
+    scalings = {'m': 1., 'cm': 100., 'mm': 1000.}
+    coordsystem_fname = _find_matching_sidecar(bids_fname, bids_root,
+                                               suffix='coordsystem.json')
     electrodes_fname = _find_matching_sidecar(bids_fname, bids_root,
                                               "electrodes.tsv",
                                               allow_fail=True)
+    orig_electrodes_dict = _from_tsv(electrodes_fname,
+                                     [str, float, float, float, str])
+    for coord_unit, scaling in scalings.items():
+        # update coordinate SI units
+        _update_sidecar(coordsystem_fname, 'iEEGCoordinateUnits', coord_unit)
+        electrodes_dict = _from_tsv(electrodes_fname,
+                                    [str, float, float, float, str])
+        for axis in ['x', 'y', 'z']:
+            electrodes_dict[axis] = \
+                np.multiply(orig_electrodes_dict[axis], scaling)
+        _to_tsv(electrodes_dict, electrodes_fname)
+
+        # read in raw file w/ updated montage
+        raw_test = read_raw_bids(bids_fname, bids_root)
+
+        # obtain the sensor positions and make sure they're the same
+        assert_dig_allclose(raw.info, raw_test.info)
+
+    # check that coordinate systems can be used and defaults to mri
+    coordinate_frames = ['lia', 'ria', 'lip', 'rip', 'las']
+    unknown_coord_frame = _IEEG_COORDINATE_FRAME_DICT['unknown']
+    head_coord_frame = _IEEG_COORDINATE_FRAME_DICT['head']
+    for coord_frame in coordinate_frames:
+        # update coordinate units
+        _update_sidecar(coordsystem_fname, 'iEEGCoordinateSystem', coord_frame)
+        # read in raw file w/ updated coordinate frame
+        # and make sure all digpoints are MRI coordinate frame
+        with pytest.warns(UserWarning, match="Coordinate frame"):
+            raw_test = read_raw_bids(bids_fname, bids_root)
+            # XXX: add test for 'unknown' possibly
+            for digpoint in raw_test.info['dig']:
+                assert digpoint['coord_frame'] in [unknown_coord_frame,
+                                                   head_coord_frame]
+
+    # coordinate frames in mne-python should all map correctly
+    coordinate_frames = ['mri', 'mri_voxel', 'mni_tal', 'ras', 'fs_tal']
```

Comment:
do you write the MNE-Python coordinate frame to BIDS?

They seem to have a different set of names: https://bids-specification.readthedocs.io/en/stable/99-appendices/08-coordinate-systems.html no?

Reply:
we should

Reply:
This now done via the `space` entity and a combo w/ `XXXCoordinateSystem = 'Other'`, which imo... is not the best way to do it, but it's what I'm getting from the BIDS specification.